components/ws-manager/pkg/manager/metrics.go

```diff
@@ -360,3 +361,45 @@ func (vec *timeoutSettingsVec) Collect(ch chan<- prometheus.Metric) {
 		ch <- metric
 	}
 }
+
+func newSubscriberQueueLevelVec(m *Manager) *subscriberQueueLevelVec {
+	name := prometheus.BuildFQName(metricsNamespace, metricsWorkspaceSubsystem, "subscriber_queue_level")
+	return &subscriberQueueLevelVec{
+		Name: name,
+		Desc: prometheus.NewDesc(
+			name,
+			"Current fill level of the subscriber queue",
+			[]string{"client"},
+			prometheus.Labels(map[string]string{}),
+		),
+		M: m,
+	}
+}
+
+type subscriberQueueLevelVec struct {
+	Name string
+	Desc *prometheus.Desc
```

Comment:
Could we transform that into a [prometheus.Gauge](https://github.com/prometheus/client_golang/blob/b8fcd28885584356c562e9ecad42cbec45f4ce96/prometheus/gauge.go#L66-L76) instead of prometheus.Desc?

I'd expect that we can we use the [Gauge functions](https://github.com/prometheus/client_golang/blob/b8fcd28885584356c562e9ecad42cbec45f4ce96/prometheus/gauge.go#L36-L49) to set the values of this metric, i.e., `add()` or `inc()` when the queue increases and `sub()` or `dec()` when the queue decreases.

Reply:
I had originally refrained from using a `GaugeVec`, thinking it'd be more effort (especially the label management). Turns out, it's not :) Thanks @ArthurSens for prompting this change 🚀 

W.r.t. to the client IP, we cannot really get ahold of the pod name easily. Also, you can have multiple subscription requests from the same client, hence the port and time for making them unique. I'd fret from leaving them out of the metric, but we can do some clever label mapping stuff if we care about the number of concurrent connections per client.